client/app/components/ApplicationArea/ErrorMessage.jsx

```diff
@@ -14,19 +14,29 @@ function getErrorMessageByStatus(status, defaultMessage) {
   }
 }
 
-export function getErrorMessage(error) {
-  const message = "It seems like we encountered an error. Try refreshing this page or contact your administrator.";
+export function getErrorMessage(
+  error,
+  defaultMessage = "It seems like we encountered an error. Try refreshing this page or contact your administrator."
+) {
   if (isObject(error)) {
     // HTTP errors
     if (error.isAxiosError && isObject(error.response)) {
-      return getErrorMessageByStatus(error.response.status, get(error, "response.data.message", message));
+      const errorData = get(error, "response.data", {});
+
+      // handle cases where the message is an object as { "message": msg } or { "error": msg }
+      const errorMessage = errorData.message || errorData.error || defaultMessage;
+      return getErrorMessageByStatus(error.response.status, errorMessage);
```

Comment:
@kravets-levko I created the test and added `data.error` to also be evaluated (it was previously handled in `PromiseRejectionError`).